src/main/java/seedu/intrack/model/internship/UniqueInternshipList.java

```diff
@@ -186,14 +186,14 @@ public void dateSortDescending() {
      * Sorts all the {@code Internship} in the list by their respective salaries in ascending order
      */
     public void salarySortAscending() {
-        internalList.sort(Comparator.comparing(o -> o.getSalary().value));
+        internalList.sort(Comparator.comparing(o -> Integer.parseInt(o.getSalary().value)));
```

Comment:
Good job fixing this! Would be good to change some of the test data to Strings of different lengths so we can test for this.